pkg/config/config.go

```diff
@@ -161,6 +160,9 @@ type ApiServer struct {
 	// AdvertiseAddress for endpoint slices in kubernetes service. Developer
 	// only parameter, wont show in show-config commands or docs.
 	AdvertiseAddress string `json:"advertiseAddress,omitempty"`
+	// SkipInterfaceConfiguration determines whether the loopback interface
+	// should be assigned the AdvertiseAddress configured before.
+	SkipInterfaceConfiguration bool `json:"skipInterfaceConfiguration,omitempty"`
```

Comment:
Why do we want to make the user make this choice? Is this something we can figure out on our own?

Reply:
Not really, this is a complex. You have all these possibilities:
- Custom AdvertiseAddress + SkipInterfaceConfiguration=true: This means the api server is either in a different node or is using an already configured interface.
- Custom AdvertiseAddress + SkipInterfaceConfiguration=false: This is deliberately ignoring the default next_subnet_after_service_cidr ip, as there might be collisions with that range or simply put it is required to have a different subnet. It does configure the lo interface with the ip.
- No AdvertiseAddress + SkipInterfaceConfiguration=true: Same as having a custom advertise address, there might be an interface already configured with the first ip from next_subnet_after_service_cidr.
- No AdvertiseAddress + SkipInterfaceConfiguration=false: Default everything, it will configure the first valid ip from next_subnet_after_service_cidr in lo interface. This will be the common case.

Reply:
Should we make the `SkipInterfaceConfiguration` an implicit (devel-only) option and assume it is false for single node deployment? because it seems `SkipInterfaceConfiguration=true` is for multi-node consideration.

Reply:
Multi node does require this option, however you could also have an IP range you can use in the node and there is no need to configure it for lo interface. I wonder if this is something that could happen.
Also, it defaults to false: https://github.com/openshift/microshift/pull/1478/files#diff-a3d824da3c42420cd5cbb0a4a2c0e7b5bfddd819652788a0596d195dc6e31fa5R251

Reply:
I'm still unclear on why we need the boolean at all. If the user provides a custom address, we can say that they must pre-configure the interface to use with that address, even if they just use the loopback interface. If they do not provide an address, then we will configure the loopback interface with an address we choose.

Reply:
So are we ok to say that if not using default then the configuration side of the interface lies on the admin? Having in mind that custom here means "anything different than "service-CIDR-next-subnet". If that is ok then I am totally fine to remove the exposed option!

Reply:
> So are we ok to say that if not using default then the configuration side of the interface lies on the admin? Having in mind that custom here means "anything different than service-CIDR-next-subnet". If that is ok then I am totally fine to remove the exposed option!

Yes, I think that's consistent with how we separate responsibilities for the OS setup in other cases.